### PR TITLE
test: Set non-null secret reader for tests

### PR DESCRIPTION
**Pull request by jkosh44**
### Motivation
This PR fixes a previously unreported bug.

### Testing

- [X] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - N/A

### CODE REVIEW

File: src/storage/src/client/connections.rs [hunk 1]
```diff
@@ -134,6 +134,13 @@ impl ConnectionContext {
             secrets_reader,
         }
     }
+    /// Sets the `secrets_reader` field on `self`.
+    ///
+    /// Returns `Self`.
+    pub fn secrets_reader(mut self, secrets_reader: Arc<dyn SecretsReader>) -> ConnectionContext {
+        self.secrets_reader = secrets_reader;
+        self
+    }
```
Comment (benesch) [inline src/storage/src/client/connections.rs:143]: Ah! Can you instead delete the `Default` impl, and add one method like:

```
impl ConnectionContext
    fn for_tests(secrets_reader: Arc<dyn SecretsReader>) {
        // The rest of the old Default impl.
    }
}
```

I'm 95% sure that `Default` impl was only ever used for tests; probs best just to make the correct usage in tests explicit.

**Comment by benesch** [comment]
I _think_ you can probably outright delete the `NullSecretsReader` too?

File: src/storage/src/client/connections.rs (hunk 1 above)
Comment (jkosh44) [reply src/storage/src/client/connections.rs]: Done

**Comment by jkosh44** [comment]
> I _think_ you can probably outright delete the `NullSecretsReader` too?

Yep, just removed it.

**Review by benesch — APPROVED** [review]
Wonderful thank you!

**Review by cjubb39 — APPROVED** [review]
lgtm!

**Comment by benesch** [comment]
Ah, right. No good deed goes unpunished. You'll need to update the definition of `Storaged` and `Computed` to pass the `--secrets-data-dir` argument as `/mzdata/secrets` or whatever it is.

**Pull request merged**